src/rdkafka.h

```diff
@@ -1214,11 +1214,29 @@ RD_EXPORT size_t rd_kafka_header_cnt (const rd_kafka_headers_t *hdrs);
  * @brief Configuration result type
  */
 typedef enum {
-	RD_KAFKA_CONF_UNKNOWN = -2, /**< Unknown configuration name. */
-	RD_KAFKA_CONF_INVALID = -1, /**< Invalid configuration value. */
-	RD_KAFKA_CONF_OK = 0        /**< Configuration okay */
+	RD_KAFKA_CONF_SCOPE_MISMATCH = -3, /**< Invalid scope for consumer or
```

Comment:
I don't think this added enum is strictly required, using UNKNOWN and providing a descriptive error string is sufficient and avoids applications having to deal with another return value.

Reply:
I feel that UNKNOWN is too broad of an error and is misleading. If someone checks for it, and sees UNKNOWN, they will think the option does not exist which is even worse IMHO. Also most user facing APIs don't have an error string passed in. I do set errno however.